core_contracts/governance/governance.py

```diff
@@ -90,34 +173,53 @@ def cancelVote(self, name: str) -> None:
         proposal.status.set(ProposalStatus.STATUS[ProposalStatus.CANCELLED])
 
     @external
-    @only_owner
-    def defineVote(self, name: str, description: str, quorum: int, vote_start: int, duration: int, snapshot: int, actions: str,
-                   majority: int = MAJORITY) -> None:
+    def defineVote(self, name: str, description: str, vote_start: int, duration: int, 
+                   snapshot: int, actions: str) -> None:
         """
-        Names a new vote, defines quorum, and actions.
+        Define a new vote and which actions are to be executed if it is successful.
+
+        Parameters:
+        name         -   Name of the vote.
+        description  -   Description of the vote.
+        vote_start   -   Day to start the vote.
+        duration     -   Number of days the vote will be active.
+        snapshot     -   Which day to use for the baln stake snapshot.
+        actions      -   Json string on the form:  {'<action_1>': {<kwargs for action_1>},
+                         {'<action_2>': {kwargs_for_action_2}}, ...}
         """
         if len(description) > 500:
             revert(f'Description must be less than or equal to 500 characters.')
-        if not 0 < quorum < 100:
-            revert(f'Quorum must be greater than 0 and less than 100.')
-        if vote_start <= self.getDay():
+        if vote_start < self.getDay():
             revert(f'Vote cannot start before the current time.')
-        if snapshot >= vote_start:
-            revert(f'Snapshot reference index must be less than vote start.')
+        if not self.getDay() <= snapshot <= vote_start:
+            revert(f'The reference snapshot must be in the range: [current_day ({self.getDay()}), '
```

Comment:
I was not sure if we necessarily had to restrict the snapshot to be the current day or later. A snapshot from a previous day could be valid, and it might make sense to allow the most recently completed snapshot, although I don't see any cases where an earlier one would be justified. In any case there has to be some limit. That most recently completed snapshot may actually be the best choice. Makes me wonder if we ought to provide defaults for snapshot, duration and vote_start.

Reply:
I agree that an earlier snapshot than current day might make sense in some rare cases. Perhaps in a senario where only users who used the platform at an earlier date are affected by some vote. Can't think of anything concrete right now though. And as you said, there has to be some limit. Not sure about this one.

Yeah, good question. If we go with defaults; would it make more sense to handle that via the frontend of backend?

Reply:
I guess I'd be good with just allowing the most recent completed snapshot or later for the reference.

For defaults I have tried to supply sensible defaults on contracts, but I guess it only really helps to make the frontend code a little cleaner, or maybe helps us if we are calling the methods directly. In general defaults could all be handled in the front end. That's probably good enough. So, yeah, we can skip adding defaults.

Reply:
Alright. Sounds good.